core/src/hash_table/links_entry.rs

```diff
@@ -0,0 +1,204 @@
+}
+
+impl LinkListEntry {
+    pub fn new(links: &[Link]) -> Self {
+        LinkListEntry {
+            links: links.to_vec(),
+        }
+    }
+}
+
+impl ToEntry for LinkListEntry {
+    // Convert a LinkEntry into a JSON array of Links
+    fn to_entry(&self) -> Entry {
+        let json_array = serde_json::to_string(self).expect("LinkListEntry should serialize");
+        Entry::new(EntryType::LinkList.as_str(), &json_array)
+    }
+
+    fn new_from_entry(entry: &Entry) -> Self {
+        assert!(EntryType::from_str(&entry.entry_type()).unwrap() == EntryType::LinkList);
+        serde_json::from_str(&entry.content()).expect("entry is not a valid LinkListEntry")
+    }
+}
+//-------------------------------------------------------------------------------------------------
+// Tests
+//-------------------------------------------------------------------------------------------------
+#[cfg(test)]
+pub mod tests {
+    extern crate test_utils;
+    use super::*;
+    use action::{Action, ActionWrapper};
+    use hash_table::sys_entry::{EntryType, ToEntry};
+    use instance::{tests::test_context, Instance, Observer};
+    use std::{str::FromStr, sync::mpsc::channel};
+    /// Committing a LinkEntry to source chain should work
+    #[test]
+    fn can_commit_link() {
+        // Create Context, Agent, Dna, and Commit AgentIdEntry Action
+        let context = test_context("alex");
+        let link = Link::new("12", "34", "fake");
+        let link_entry = LinkListEntry::new(&[link]);
+        let commit_action = ActionWrapper::new(Action::Commit(link_entry.to_entry()));
+        // Set up instance and process the action
+        let instance = Instance::new();
+        let state_observers: Vec<Observer> = Vec::new();
+        let (_, rx_observer) = channel::<Observer>();
+        instance.process_action(commit_action, state_observers, &rx_observer, &context);
+        // Check if LinkEntry is found
+        assert_eq!(1, instance.state().history.iter().count());
+        instance
+            .state()
+            .history
+            .iter()
+            .find(|aw| match aw.action() {
+                Action::Commit(entry) => {
+                    assert_eq!(
+                        EntryType::from_str(&entry.entry_type()).unwrap(),
+                        EntryType::LinkList,
+                    );
+                    assert_eq!(entry.content(), link_entry.to_entry().content());
+                    true
+                }
+                _ => false,
+            });
+    }
+    /// Committing a DnaEntry to source chain should work
+    #[test]
+    fn can_commit_multilink() {
+        // Create Context, Agent, Dna, and Commit AgentIdEntry Action
+        let context = test_context("alex");
+        let link1 = Link::new("12", "34", "fake");
+        let link2 = Link::new("56", "78", "faux");
+        let link3 = Link::new("90", "ab", "fake");
+        let link_entry = LinkListEntry::new(&[link1, link2, link3]);
```

Comment:
The LinkEntry in the source-chain is just for one link, but the metadata on the base entry is stored per tag in the hash table since the API gets Links per tag and not per target entry, So the metadata points to a LinkListEntry storing all the links for that tag.

Reply:
Just as a record here for people reading the PR.  @ddd-mtl and I talked about how part of the confusion is that we are using the same HT implementation for chain and dht, and "entry" could mean anything you put in an HT.  But, actually we need to distinguish those things and types that are chain-entries because they are what's in the DNA and they are the only things you can link on.  So, though we are adding this PR for now, the LinkListEntry is NOT something that you can commit to the chain, it's just data you get back from a DHT query.